### PR TITLE
Use RunInference to perform custom inference in custom_inference notebook

### PR DESCRIPTION
**Pull request by AnandInguva**
Fixes: https://github.com/apache/beam/issues/29297

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by AnandInguva** [comment]
R: @damccorm @liferoad

**Comment by liferoad** [comment]
The changes look good to me. Two suggestions:

- Can we keep the old way as well? It is a good code snippet for users to learn Beam DoFn
- Can we link the VertexAIModelHandler notebook inside this?

**Comment by damccorm** [comment]
> Can we keep the old way as well? It is a good code snippet for users to learn Beam DoFn

I would vote against this. There are other examples of using a DoFn (e.g. https://github.com/apache/beam/blob/e612060217120329d415b44f9ecdd7d0b347a21d/examples/notebooks/documentation/transforms/python/elementwise/pardo-py.ipynb#L81), but the preferred way for people to do remote inference is with RunInference.

**Comment by liferoad** [comment]
The current example is too simple and quite useless. I even think we should not just keep the old way but also use it to show why they should switch to `RunInference`.

**Comment by AnandInguva** [comment]
> The changes look good to me. Two suggestions:
> 
> * Can we keep the old way as well? It is a good code snippet for users to learn Beam DoFn
> * Can we link the VertexAIModelHandler notebook inside this?

**Review by liferoad — APPROVED** [review]

File: examples/notebooks/beam-ml/custom_remote_inference.ipynb [hunk 1]
```diff
@@ -60,6 +60,7 @@
         "To reduce the number of steps that you need to take, RunInference supports features like batching. For more infomation about the RunInference API, review the [RunInference API](https://beam.apache.org/releases/pydoc/current/apache_beam.ml.inference.html#apache_beam.ml.inference.RunInference),\n",
         "which demonstrates how to implement model inference in PyTorch, scikit-learn, and TensorFlow.\n",
         "\n",
+        "There is [VertexAIModelHandlerJson](https://github.com/apache/beam/blob/1c24bc94c9ae5cf93dc129e657964858d4aafc67/sdks/python/apache_beam/ml/inference/vertex_ai_inference.py#L61) which is used to make remote inference calls to VertexAI. In this notebook, we will make custom `ModelHandler` to do remote inference calls using CloudVision API.\n",
```
Comment (liferoad) [inline examples/notebooks/beam-ml/custom_remote_inference.ipynb:63]: do we use master or a commit hash?

Comment (AnandInguva) [reply examples/notebooks/beam-ml/custom_remote_inference.ipynb]: Modified it to master

**Pull request merged**